test/saithrift/tests/sail3.py

```diff
@@ -1477,3 +1477,194 @@ def runTest(self):
+                                            ip_ttl=63)
+                    masked_exp_pkt = Mask(exp_pkt)
+                    masked_exp_pkt.set_do_not_care_scapy(ptf.packet.Ether,"dst")
+
+                    send_packet(self, 0, str(pkt))
+                    (match_index,rcv_pkt) = verify_packet_any_port(self,masked_exp_pkt,destanation_ports)
+                    logging.debug("found expected packet from port %d" % destanation_ports[match_index])
+                    pkt_counter[match_index] += 1
+                    sport = random.randint(0,0xffff)
+                    dport = random.randint(0,0xffff)
+                        
+        #final uniform distribution check
+        logging.debug(pkt_counter)
+        logging.debug(lag_size)
+        lag_packets = sum(pkt_counter[1:lag_size])
+        lag_average = lag_packets/(len(self.lag_members) + 1)
+        logging.debug("the sum of packets through the lag is " + str(lag_packets))
+        logging.debug("the lag average for the lag is " + str(lag_average))
+        for stat_port in range(1,lag_size):
+            logging.debug( "PORT #"+str(stat_port)+":")
+            logging.debug(str(pkt_counter[stat_port]))
+            self.polarizationCheck(pkt_counter[stat_port],lag_average)
+        rifs_average = sum(pkt_counter)/(len(self.port_rifs) + 1)
+        logging.debug("lag average " + str(lag_average))
+        self.polarizationCheck(lag_packets,rifs_average)
+        for stat_port in range(lag_size,self.total_changing_ports):
+            logging.debug( "PORT #"+str(stat_port)+":")
+            logging.debug(str(pkt_counter[stat_port]))
+            self.polarizationCheck(pkt_counter[stat_port],rifs_average)
+    
+                 
+    def runTest(self):
+        """
+        For sai server, testing different lags with router
+        ---- Test for 16 ports minimun ----
+        Steps
+        1. Create virtual router, and rif for src port
+        2. create a lag and lag rif,add ports to the lag and the rest of the ports connect to rifs
+        3. configure neighbors, nhops for all of the rifs
+        4. make ecmp route with all of the nhops
+        5. send packets from src port
+        6. check polarization check in the lag and in the ecmp
+        7. remove rifs, neighbors, nhops, lag members, lag and route
+        8. repeat steps 3-7 with differnt numbers of lag members and rifs
+        8. clean up.
+        """
+       
+          
+        print
+        print "L3MultipleEcmpLagTest"
+        #general configuration 
+        random.seed(1)
+        switch_init(self.client)
+        self.src_port = port_list[0]
+
+        self.vr_id = sai_thrift_create_virtual_router(self.client, self.v4_enabled, self.v6_enabled)
+        rif_port_id = sai_thrift_create_router_interface(self.client, self.vr_id, 1, self.src_port, 0, self.v4_enabled, self.v6_enabled, '')
+        
+        try:
+            for lag_size in range(self.first_changing_port,self.total_changing_ports):
```

Comment:
its not the min lag size
its the first changing port
the min lag size is lag with only one port
so the first "changing port" is 2